Shared/sdk/SharedUtil.File.hpp

```diff
@@ -1052,9 +1066,55 @@ int SharedUtil::File::Delete(const char* szFilename)
 
 int SharedUtil::File::Rename(const char* szOldFilename, const char* szNewFilename)
 {
-#ifdef WIN32
+#ifdef _WIN32
     return _wrename(FromUTF8(szOldFilename), FromUTF8(szNewFilename));
 #else
     return rename(szOldFilename, szNewFilename);
 #endif
 }
+
+std::vector<std::string> SharedUtil::ListDir(const char* szPath)
+{
+    std::vector<std::string> entries;
+#if _HAS_CXX17
+    namespace fs = std::filesystem;
+    if (!DirectoryExists(szPath))
+        return {};
+
+    for (const auto& entry : fs::directory_iterator(szPath))
+    {
+        entries.push_back(entry.path().filename().string());
+    }
```

Comment:
This is going to include non-directory and non-regular-file file types (block file, character file, socket file, symlinks) if I am not wrong. How about we only allow `entry.is_regular_file()` and `entry.is_directory()` here?

Also, you're using the throwing filesystem functions.

Reply:
> This is going to include non-directory and non-regular-file file types (block file, character file, socket file, symlinks) if I am not wrong. How about we only allow `entry.is_regular_file()` and `entry.is_directory()` here?
> 
> Also, you're using the throwing filesystem functions.

sure, we can do that. With throwing we can just use `try-catch` here